Gems/PhysX/PhysX5/Source/Module.cpp

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/std/smart_ptr/unique_ptr.h>
+#include <AzCore/Module/Module.h>
+#include <AzCore/Module/DynamicModuleHandle.h>
+#include <AzCore/Serialization/SerializeContext.h>
+#include <AzCore/Component/ComponentApplicationBus.h>
+
+#include <Configuration/PhysXSettingsRegistryManager.h>
+#include <Source/SystemComponent.h>
+#include <ComponentDescriptors.h>
+
+#if defined(PHYSX_EDITOR)
+#include <Source/EditorComponentDescriptors.h>
+#include <Editor/Source/Components/EditorSystemComponent.h>
+#include <Editor/Source/Configuration/PhysXEditorSettingsRegistryManager.h>
+#endif // defined(PHYSX_EDITOR)
+
+#include <PhysX_Traits_Platform.h>
+
+#include <System/PhysXSystem.h>
+#include <System/PhysXCookingParams.h>
+
+namespace PhysX
+{
+    class Module
+        : public AZ::Module
+    {
+    public:
+        AZ_RTTI(PhysX::Module, "{A2E6D801-1510-4EFD-B40A-8C3B066A0323}", AZ::Module);
+        AZ_CLASS_ALLOCATOR(PhysX::Module, AZ::SystemAllocator)
+
+        Module()
+            : AZ::Module()
+#if defined(PHYSX_EDITOR)
+            , m_physXSystem(AZStd::make_unique<PhysXEditorSettingsRegistryManager>(), PxCooking::GetEditTimeCookingParams())
+#else
+            , m_physXSystem(AZStd::make_unique<PhysXSettingsRegistryManager>(), PxCooking::GetRealTimeCookingParams())
+#endif
+        {
+            // PhysXSystemConfiguration needs to be 16-byte aligned since it contains a SIMD vector4.
+            // The vector4 itself is aligned relative to the module class, but if the module class is
+            // not also aligned, it will crash. This checks makes sure they will be aligned to 16 bytes.
+            static_assert(alignof(PhysX::PhysXSystemConfiguration) == 16);
+            static_assert(alignof(PhysX::PhysXSystem) == 16);
+            static_assert(alignof(PhysX::Module) == 16);
```

Comment:
I believe this static_assert needs to be outside of the `Module` class, since asking for the `sizeof/alignof` a class inside of it's definition can't be done until the after the definition is complete.

Reply:
This was a duplication of the module specific source for the original PhysX. Interestingly enough, this assert hasn't been failing in the builds.